client/src/components/create-recipe/CreateForm.test.js

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react';
+import { render, fireEvent, act, getAllByText } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import AuthService from '../../service/AuthService';
+import RecipeService from '../../service/RecipeService';
+import AuthProvider from '../../context/AuthContext';
+import CreateForm from './CreateForm';
+import ImageCompressor from '../Utils/ImageCompressor';
+import ToBase64 from '../Utils/ToBase64';
+
+jest.mock('../../service/AuthService');
+jest.mock('../Utils/ImageCompressor');
+jest.mock('../Utils/ToBase64');
+jest.mock('../../service/RecipeService');
+
+const mockHistoryPush = jest.fn();
+global.URL.createObjectURL = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+      push: mockHistoryPush,
+    }),
+  }));
+
+test('renders content correctly', async () => {
+    const authenticatedResponse = {isAuthenticated: true, user: {username: 'Bpun1p'}};
+    AuthService.isAuthenticated.mockResolvedValue(authenticatedResponse);
+
+    const { getByTestId, getAllByTestId, getByText } = render(<CreateForm />, {wrapper: AuthProvider});
+
+    await act(() => Promise.resolve());
```

Comment:
It's a hacky way to write 'wait a bit'. That it uses act makes it take component state changes into account. It targets specifically promises because it uses a promise and resolves right after instantly resolved promises in the component took action

Reply:
Hmm I don't follow, maybe we can have a call about this

Reply:
adjustments made.